### DISCUSSION TITLE
Trait Validation

### DISCUSSION THREAD

**Discussion by kylewallace** [livewire/livewire#3396] [1. Help]
Hi all,

Is anyone able to provide an example of how they handle additional validation rules from traits into a Livewire component?

I was hoping there would be similar functionality to that mentioned in the traits page on the official docs (i.e. `mountTraitName`, `updatedTraitName`) so something like `rulesTraitName` but unfortunately there isn't?

I've tried using the `mountTraitName` method and `array_merge` to push a `$traitRules` property into the `$rules` array but I think I am going about it wrong.

Any assistance or examples would be appreciated!

Thanks.

**Comment by martinsnajdr** [comment]
This should be considered as a feature. Would be very useful.

Right now I'm merging a rules array from trait inside rules function in Livewire component manually. It would be nice if Livewire was looking for rulesTraitName and merge it automatically.
